### DISCUSSION TITLE
Livewire v3 support

### DISCUSSION THREAD

**Discussion by gizzmojo** [joshhanley/livewire-autocomplete#23] [Help] [answered]
Wondered if there was a plan on upgrading this package to support Livewire v3?

I attempted to update it but I stumbled across an issue where `$dispatch((this.name ?? 'autocomplete') + '-input', this.decoupledValue)` wasn't making any request.

**Comment by Edgarborras94** [comment]
I've been trying to update the package by myself but ending with getting the results variable with an array of null objects...

maybe @joshhanley  can illuminate a bit what's goin on in the new v3 hehe...

chunk of code of the autocomplete.blade.php component updated from upgrade guide v3:

```
x-data="autocomplete({
        name: '{{ $name }}',
        decoupledValue: null,
        focusAction: '{{ $focusAction->value ?? null }}',
        value: $wire.$entangle('{{ $inputProperty->value }}', true),
        results: $wire.$entangle('{{ $resultsProperty->value }}', true),
        selected: $wire.$entangle('{{ $selectedProperty->value }}', true),
        idAttribute: '{{ $getOption('id') }}',
        searchAttribute: '{{ $getOption('text') }}',
        autoSelect: {{ $autoSelect ? 'true' : 'false' }},
        allowNew: {{ $allowNew ? 'true' : 'false' }},
        loadOnceOnFocus: {{ $loadOnceOnFocus ? 'true' : 'false' }},
    })"
```

I've also upgrade some other code but no luck with this `$wire.$entangle` thing... I'm surely missing something.

I have created a fork in my github account, if anyone want to take a closer look...

**Comment by joshhanley** [comment]
@gizzmojo yep, I do plan to upgrade it, just been busy working on Livewire V3 at the moment.

@Edgarborras94 thanks for giving it a go! It's possible some of the Livewire methods I use to get the data aren't compatible with V3 (as they weren't documented methods). Also this autocomplete was built with using Eloquent Collection binding in mind, which we don't recommend using in V3.

I have been working on a new version of this package, which I think will work with V3 without any changes and isn't centred around Eloquent Collections. But the problem is, it's a different API, so I'm not sure yet how to version it, being that I never actually tagged V1 of this package.

That being said if we can easily get this version of it compatible with V3, then I'll happily tag a new version with V3 support.

**Reply by Edgarborras94** [reply]
I see... maybe tomorrow I will give it a try taking that into account :)

cheers!

**Reply by joshhanley** [reply]
@Edgarborras94 ok thanks! 🙂 if you have any issues, let me know what you get stuck on and I'll see if I can provide any insight.

**Reply by Edgarborras94** [reply]
@joshhanley OK! I think I got it working: https://github.com/Edgarborras94/livewire-autocomplete should I do a new branch for the PR? I have done some manual testing and everthing seems fine. I also have the autocomplete demo updated for V3, maybe I could make also a PR after this one

EDIT:
Some screenshots: 
![image](https://github.com/joshhanley/livewire-autocomplete/assets/6952403/fbbee6de-51c9-4e8a-9632-60cd889b57a3)
![image](https://github.com/joshhanley/livewire-autocomplete/assets/6952403/767fe473-a674-43d1-9aee-cba5fffe76d5)
![image](https://github.com/joshhanley/livewire-autocomplete/assets/6952403/d3dde9d2-5b6c-41b0-be7b-659cb06e34c2)

**Reply by joshhanley** [reply]
@Edgarborras94 Legend! Thanks! Yep a new branch would be great as it means I can work on the pr if there are any changes needed 🙂

**Reply by Edgarborras94** [reply, marked as answer]
PR done: #24 I hope everything goes well!!
